### PR TITLE
`BadgeCount` - Convert to TypeScript

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

Converts the `BadgeCount` component to TS

### :link: External links

Jira ticket: https://hashicorp.atlassian.net/browse/HDS-2684

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/src/components/hds/badge-count/index.ts [hunk 1]
```diff
@@ -6,14 +6,35 @@
...
+export const COLORS: string[] = Object.values(HdsBadgeCountColorValues);
+export const DEFAULT_SIZE = HdsBadgeCountSizeValues.Medium;
+export const DEFAULT_TYPE = HdsBadgeCountTypeValues.Filled;
+export const DEFAULT_COLOR = HdsBadgeCountColorValues.Neutral;
+
+interface HdsBadgeCountIndexSignature {
+  Args: {
+    size: HdsBadgeSizes;
+    type: HdsBadgeTypes;
+    color: HdsBadgeCountColors;
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply packages/components/src/components/hds/badge-count/index.ts:31]: yes, optional arguments should be marked as such – the codemod doesn't have such insights. I will make sure to add this to the docs

File: packages/components/src/components/hds/badge-count/index.ts [hunk 2]
```diff
@@ -6,14 +6,35 @@
...
+import {
+  HdsBadgeCountSizeValues,
+  HdsBadgeCountTypeValues,
+  HdsBadgeCountColorValues,
+} from './types.ts';
+import type {
+  HdsBadgeSizes,
+  HdsBadgeTypes,
+  HdsBadgeCountColors,
+} from './types.ts';
```
Comment (alex-ju) [inline packages/components/src/components/hds/badge-count/index.ts:18]: Personally not as opinionated on this, but I remember you were keen on [getting these ordered alphabetically](https://github.com/hashicorp/design-system/pull/1991#discussion_r1524606556).

**Review by alex-ju — APPROVED** [review]

File: packages/components/src/components/hds/badge-count/index.ts (hunk 2 above)
Comment (didoo) [reply packages/components/src/components/hds/badge-count/index.ts:18]: Let's add it to the list of things we want to standardize across components (eg. order of the Args: "logical"-ish, like it appears in the Component API docs, or in the getters order, or more simply alphabetical?)

Comment (didoo) [reply packages/components/src/components/hds/badge-count/index.ts:18]: For now I'll sort these alphabetically.

**Pull request merged**